artcommon/artcommonlib/konflux/konflux_db.py

```diff
@@ -152,11 +152,54 @@ async def search_builds_by_fields(self, start_search: datetime, end_search: date
         self.logger.debug('Found %s builds', results.total_rows)
         return results
 
+    async def search_builds_by_fields_with_window(
+            self,
+            start_search: typing.Optional[datetime] = None,
+            end_search: typing.Optional[datetime] = None,
+            search_window: int = 90,
+            where: typing.Optional[typing.Dict[str, typing.Any]] = None,
+            extra_patterns: typing.Optional[typing.Dict] = None,
+            order_by: str = '',
+            sorting: str = 'DESC',
+            limit: typing.Optional[int] = None) -> typing.AsyncGenerator[Row, None]:
+        """ Execute search_builds_by_fields() with a window of given days.
+
+        :param start_search: The start time of the search (inclusive). If None, the search starts 3 years ago.
+        :param end_search: The end time of the search (exclusive). If None, the search ends now.
+        :param search_window: The number of days in the search window. Default is 90 days.
+        :param where: The WHERE clause of the search.
+        :param extra_patterns: The extra patterns of the search.
+        :param order_by: The column to order by.
+        :param sorting: The sorting order.
+        :param limit: The limit of the search. None for no limit.
+        :return: The search results.
+        """
+        # Table is partitioned by start_time. Perform an iterative search within given days of windows, going back to start_search
+        # at most. This will let us reduce the amount of scanned data (and the BigQuery usage cost), as in the vast
+        # majority of cases we would find a build in the first search_window interval.
+        if start_search and end_search and start_search >= end_search:
+            raise ValueError(f"start_search {start_search} must be earlier than end_search {end_search}")
+        end_search = end_search or datetime.now(tz=timezone.utc)
+        start_search = start_search or end_search - timedelta(days=365*3)
```

Comment:
`get_latest_build()` goes back to 1 years at most (12 windows * 30 days). We should define a unique threshold for all searches and store it in a centralized place as a constant.